torch/fx/subgraph_rewriter.py

```diff
@@ -44,7 +44,26 @@ def attributes_are_equal(pn : Node, gn : Node) -> bool:
             if (pn.op == "placeholder"
                     or (pn.op == "output" and gn.op != "placeholder")):
                 return True
-            return pn.op == gn.op and pn.target == gn.target
+            if pn.op != gn.op:
+                return False
+            # Fuzzy matching for root-relative targets
+            if pn.op == "get_attr" or pn.op == "call_module":
+                # Consider a target "suffix" everything from the start
+                # of the rightmost alphabetical string ([a-zA-Z]) to the
+                # end of the target. For example, `layer4.2.conv2`,
+                # `3.conv2`, and `conv2` will all have the same suffix,
+                # but they won't be confused with `conv1`.
+                def get_target_suffix(s: str) -> str:
+                    first_letter = False
+                    for i, c in enumerate(s[::-1]):
+                        if first_letter and not c.isalpha():
+                            return s[-i:]
+                        elif c.isalpha():
+                            first_letter = True
```

Comment:
What happens if we fall off the end here? either or both of `{p,g}n_target_suffix` could be None, and if both the check would be true. If that's desired, type the output as `Optional[str]` and return None here, otherwise throw here because an invariant has not been met

Reply:
I decided to return the full string here if we fall off the end, considering some degenerate case in which there were no alphabetical characters in the target.